docs/learning/WriteAConnector.md

```diff
@@ -1,108 +1,727 @@
+  </iframe>
+</figure>
+
+Some sample queries that is helpful to debug Connector output
+
+1. Get the iModel Element id from the identifier from a given source file
+
+   ```sql
+    SELECT Element.Id FROM bis.ExternalSourceAspect WHERE Identifier='<Element Id from File>'
+   ```
+
+2. List of all known repositories (links to files in an iModel)
+
+   ```sql
+   SELECT UserLabel, Url FROM bis.repositoryLink
+   ```
+
+3. List of all physical partitions present in the iModel
+
+   ```sql
+   SELECT ECInstanceId, CodeValue from bis.physicalpartition
+   ```
+
+#### Visualizing the output
+
+To get started, please build it using the instructions provided in the [display test app README](https://github.com/imodeljs/imodeljs/blob/master/test-apps/display-test-app/README.md)
+Once the application is built and running, use the briefcase icon to open the output from the Connector. Some of the [frontend developer tools](https://www.npmjs.com/package/@bentley/frontend-devtools/v/1.5.0) are also useful to analyze the data. E.g. fdt inspect element.
+
+### Logs
+
+See this article on [Logging](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/common/Logging.md)
+
+### Error Messages
+
+See [Errors](https://www.itwinjs.org/reference/bentleyjs-core/errors/)
+
+### Building a test for a Connector
+
+For examples of Connector tests, you can review the standalone and integration tests in the Connector framework. These tests use the test framework, [mocha](https://www.npmjs.com/package/mocha) and the assertion library, [chai](https://www.npmjs.com/package/chai/v/4.2.0)
+
+## Advanced Topics
+
+### Job Subjects
+
+The FrameWork will automatically create a uniquely named Subject element in the iModel. The job subject element should be a child of the root subject and must have a unique code.
+
+A Connector is required to scope all of the subjects, definitions, and their models under its job subject element. That is,
+
+- Subjects and partitions that a connector creates should be children of the job subject element,
+- The models and other elements that the connector creates should be children of those subjects and partitions or in those models.
+
+### Schema merging
+
+Schemas that have been released for production use in end-user workflows evolve as new capabilities are added, and other improvements are made. To manage and track this schema evolution, schema versioning is used. See Schema Versioning and Generations for details on BIS's schema versioning strategy.
+
+### Units and Coordinate systems
+
+For the basics of coordinate systems in iModels, please see <https://www.itwinjs.org/learning/geolocation/>
+
+All coordinates and distances in an iModel must be stored in meters, and so the Connector must transform source data coordinates and distances into meters.
+
+For any iModel, a Connector should

```

Comment:
The lead in to the two items is weird, maybe remove the `should` and re-word it a bit or potentially re-word the list to start with "reproject the existing... " and "initializing the iModel ...".